src/qt/nametablemodel.cpp

```diff
@@ -98,7 +103,156 @@ class NameTablePriv
+                        // This name isn't ours, it's just some random name
+                        // that another user has pending.
+                        continue;
+                    }
+
+                    // The name will be ours when it confirms, but it wasn't
+                    // ours before.
+
+                    if (op == "name_firstupdate")
+                    {
+                        // This is a registration, so the reason it wasn't ours
+                        // before is that we hadn't registered it yet.
+                        status = NameTableEntry::NAME_STATUS_REGISTRATION_PENDING;
+                    }
+                    else if (op == "name_update")
+                    {
+                        // This is an update, so we weren't the one to register
+                        // it.  (If we were, there would be a confirmed entry.)
+                        // So this is an incoming transfer.
+                        status = NameTableEntry::NAME_STATUS_INCOMING_TRANSFER_PENDING;
+                    }
+                }
+                else
+                {
+                    // A confirmed entry exists.
+                    auto confirmedEntry = vNamesO[name];
+                    std::string confirmedData = confirmedEntry.value.toStdString();
+                    std::string confirmedStatus = confirmedEntry.nameStatus.toStdString();
+                    height = confirmedEntry.nHeight;
+                    expiresIn = confirmedEntry.expiresIn;
+
+                    if (confirmedStatus == NameTableEntry::NAME_STATUS_EXPIRED || confirmedStatus == NameTableEntry::NAME_STATUS_TRANSFERRED_OUT)
+                    {
+                        if (!isMine)
+                        {
+                            // This name isn't ours, it's just some random name
+                            // that another user has pending.
+                            continue;
+                        }
+
+                        // The name will be ours when it confirms, but it
+                        // wasn't ours before.
+
+                        if (op == "name_firstupdate")
+                        {
+                            // This is a registration, so the reason it wasn't
+                            // ours before is that we hadn't registered it yet.
+                            status = NameTableEntry::NAME_STATUS_REGISTRATION_PENDING;
+                        }
+                        else if (op == "name_update")
+                        {
+                            // This is an update, so we weren't the one to
+                            // register it.  (If we were, there would be a
+                            // confirmed entry.)  So this is an incoming
+                            // transfer.
+                            status = NameTableEntry::NAME_STATUS_INCOMING_TRANSFER_PENDING;
+                        }
+                    }
+                    else if (confirmedStatus == "Confirmed")
```

Comment:
Is there a reason why we have constants for all the status strings but not `"Confirmed"`?